.github/workflows/tests.yaml

```
@@ -0,0 +1,236 @@
+    name: Unit Tests for Java
+    runs-on: ['ubuntu-latest']
+    
+    needs: changes
+    if: ${{ needs.changes.outputs.run_job == 'true' }}
+    
+    steps:
+      - name: Checkout source repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
```

actionlint (1.7.7)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)